[2024-01-31T12:56:43.327+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_analysis_dag.process_stock_data manual__2024-01-31T12:46:30.226173+00:00 [queued]>
[2024-01-31T12:56:43.331+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_analysis_dag.process_stock_data manual__2024-01-31T12:46:30.226173+00:00 [queued]>
[2024-01-31T12:56:43.331+0000] {taskinstance.py:2171} INFO - Starting attempt 3 of 3
[2024-01-31T12:56:43.342+0000] {taskinstance.py:2192} INFO - Executing <Task(PythonOperator): process_stock_data> on 2024-01-31 12:46:30.226173+00:00
[2024-01-31T12:56:43.345+0000] {standard_task_runner.py:60} INFO - Started process 4167 to run task
[2024-01-31T12:56:43.346+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'stock_analysis_dag', 'process_stock_data', 'manual__2024-01-31T12:46:30.226173+00:00', '--job-id', '69', '--raw', '--subdir', 'DAGS_FOLDER/stock_analysis_dag.py', '--cfg-path', '/tmp/tmpmez5y7d_']
[2024-01-31T12:56:43.347+0000] {standard_task_runner.py:88} INFO - Job 69: Subtask process_stock_data
[2024-01-31T12:56:43.367+0000] {task_command.py:423} INFO - Running <TaskInstance: stock_analysis_dag.process_stock_data manual__2024-01-31T12:46:30.226173+00:00 [running]> on host de-airflow-vm-instance-1.europe-west10-a.c.dataengineeringfinal-411911.internal
[2024-01-31T12:56:43.402+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='onatpostac@gmail.com' AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='stock_analysis_dag' AIRFLOW_CTX_TASK_ID='process_stock_data' AIRFLOW_CTX_EXECUTION_DATE='2024-01-31T12:46:30.226173+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-01-31T12:46:30.226173+00:00'
[2024-01-31T12:56:43.407+0000] {logging_mixin.py:188} INFO - {
    "Information": "Thank you for using Alpha Vantage! Our standard API rate limit is 25 requests per day. Please subscribe to any of the premium plans at https://www.alphavantage.co/premium/ to instantly remove all daily rate limits."
}
[2024-01-31T12:56:43.407+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/deprojectue/miniconda3/envs/airflow_demo/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/deprojectue/miniconda3/envs/airflow_demo/lib/python3.9/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/deprojectue/miniconda3/envs/airflow_demo/lib/python3.9/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/deprojectue/airflow_demo/dags/stock_analysis_dag.py", line 99, in process_stock_data
    df = pd.DataFrame.from_dict(stock_data["Time Series (Daily)"], orient='index')
KeyError: 'Time Series (Daily)'
[2024-01-31T12:56:43.409+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=stock_analysis_dag, task_id=process_stock_data, execution_date=20240131T124630, start_date=20240131T125643, end_date=20240131T125643
[2024-01-31T12:56:43.418+0000] {logging_mixin.py:188} WARNING - /home/deprojectue/miniconda3/envs/airflow_demo/lib/python3.9/site-packages/airflow/utils/email.py:154 RemovedInAirflow3Warning: Fetching SMTP credentials from configuration variables will be deprecated in a future release. Please set credentials using a connection instead.
[2024-01-31T12:56:43.418+0000] {configuration.py:1046} WARNING - section/key [smtp/smtp_user] not found in config
[2024-01-31T12:56:43.418+0000] {email.py:270} INFO - Email alerting: attempt 1
[2024-01-31T12:56:43.640+0000] {email.py:281} INFO - Sent an alert email to ['onatpostac@gmail.com']
[2024-01-31T12:56:43.695+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 69 for task process_stock_data ('Time Series (Daily)'; 4167)
[2024-01-31T12:56:43.719+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-01-31T12:56:43.731+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
